manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/api/user/SourceConnectorsAPITest.java

```diff
@@ -0,0 +1,45 @@
+package com.redhat.service.smartevents.manager.v2.api.user;
+
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.redhat.service.smartevents.infra.core.api.APIConstants;
+import com.redhat.service.smartevents.manager.v2.TestConstants;
+import com.redhat.service.smartevents.manager.v2.utils.DatabaseManagerUtils;
+import com.redhat.service.smartevents.manager.v2.utils.TestUtils;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+
+import static com.redhat.service.smartevents.infra.core.api.APIConstants.USER_NAME_ATTRIBUTE_CLAIM;
+import static com.redhat.service.smartevents.manager.v2.TestConstants.DEFAULT_USER_NAME;
+import static org.mockito.Mockito.when;
+
+@QuarkusTest
+public class SourceConnectorsAPITest {
+
+    @Inject
+    DatabaseManagerUtils databaseManagerUtils;
+
+    @InjectMock
+    JsonWebToken jwt;
+
+    @BeforeEach
+    public void cleanUp() {
+        databaseManagerUtils.cleanUpAndInitWithDefaultShard();
+        when(jwt.getClaim(APIConstants.ACCOUNT_ID_SERVICE_ACCOUNT_ATTRIBUTE_CLAIM)).thenReturn(TestConstants.SHARD_ID);
+        when(jwt.containsClaim(APIConstants.ACCOUNT_ID_SERVICE_ACCOUNT_ATTRIBUTE_CLAIM)).thenReturn(true);
+        when(jwt.getClaim(APIConstants.ORG_ID_SERVICE_ACCOUNT_ATTRIBUTE_CLAIM)).thenReturn(TestConstants.DEFAULT_ORGANISATION_ID);
+        when(jwt.containsClaim(APIConstants.ORG_ID_SERVICE_ACCOUNT_ATTRIBUTE_CLAIM)).thenReturn(true);
+        when(jwt.getClaim(USER_NAME_ATTRIBUTE_CLAIM)).thenReturn(DEFAULT_USER_NAME);
+        when(jwt.containsClaim(USER_NAME_ATTRIBUTE_CLAIM)).thenReturn(true);
+    }
+
+    @Test
+    public void testGetSourceConnectorsNoAuthentication() {
+        TestUtils.listSourceConnectors().then().statusCode(401);
```

Comment:
I think it'll be a 500 at the moment.

Reply:
If the user is not authenticated it should be 401